js/background.js

```diff
@@ -43,57 +61,80 @@ function TogglePane(tab)
 
 function ProcessCommand(command)
 {
-	switch(command)
+	GetTabsToSave((returnedTabs) =>
 	{
-		case "set-aside":
-			SaveCollection();
-			break;
-		case "toggle-pane":
-			chrome.tabs.query(
-				{
-					active: true,
-					currentWindow: true
-				},
-				(tabs) => TogglePane(tabs[0])
-			)
-			break;
-	}
+		tabsToSave = returnedTabs;
+		switch(command)
+		{
+			case "set-aside":
+				SaveCollection();
+				break;
+			case "toggle-pane":
+				chrome.tabs.query(
+					{
+						active: true,
+						currentWindow: true
+					},
+					(tabs) => TogglePane(tabs[0])
+				)
+				break;
+		}
+	});
 }
 
 chrome.browserAction.onClicked.addListener((tab) =>
 {
-	chrome.storage.sync.get({ "setAsideOnClick": false }, values =>
+	GetTabsToSave((returnedTabs) =>
 	{
-		if (values?.setAsideOnClick)
-			SaveCollection();
-		else
-			TogglePane(tab);
+		tabsToSave = returnedTabs;
+
+		chrome.storage.sync.get({ "setAsideOnClick": false }, values =>
+		{
+			if (values?.setAsideOnClick)
+				SaveCollection();
+			else
+				TogglePane(tab);
+		});
 	});
 });
 
 // Adding context menu options
 chrome.contextMenus.create(
 	{
 		id: "toggle-pane",
-		contexts: ['all'],
+		contexts: ["browser_action"],
 		title: chrome.i18n.getMessage("togglePaneContext")
 	}
 );
 chrome.contextMenus.create(
 	{
 		id: "set-aside",
-		contexts: ['all'],
+		contexts: ["browser_action"],
 		title: chrome.i18n.getMessage("setAside")
 	}
 );
 
-var collections = JSON.parse(localStorage.getItem("sets")) || [];
+var collections = [];
+chrome.storage.sync.get("sets", values =>
+{
+	collections = JSON.parse(values?.sets ?? "[]");
```

Comment:
I'm not sure about this one. Values should already be the "sets" object of the storage, which should directly contain the collections.

So I think using `values.sets` (i.e, sets.sets in the storage) could be a mistake

Reply:
According to [the documentation](https://developer.chrome.com/extensions/storage#type-StorageArea) callback function parameter of the `get` is an array of requested values in any case. I know, that's weird since we can pass requested key name as a single string, but it is what it is

Reply:
You're right, I forgot about that !